src/app/helpers/attachment/attachmentLoader.ts

```diff
@@ -0,0 +1,134 @@
+import {
+    binaryStringToArray,
+    concatArrays,
+    decodeBase64,
+    decryptMessage,
+    decryptSessionKey,
+    getMessage
+} from 'pmcrypto';
+
+import { getAttachment } from '../../api/attachments';
+import { MessageExtended } from '../../models/message';
+import { Attachment, AttachmentData } from '../../models/attachment';
+import { AttachmentsCache } from '../../hooks/useAttachments';
+import { Api } from '../../models/utils';
+
+// Reference: Angular/src/app/attachments/services/AttachmentLoader.js
+
+interface SessionKey {
+    data?: Uint8Array;
+    algorithm?: string;
+}
+
+// TODO: Handle isOutside()
+
+export const decrypt = async (
+    encryptedBinaryBuffer: ArrayBuffer,
+    sessionKey: SessionKey = {}
+): Promise<AttachmentData> => {
```

Comment:
it sounds a bit counter-intuitive that the function `decryptMessage` returns an `AttachmentData` type, but I guess it should be fine

Reply:
You're right, I was too focused, it should be something generic from pmcrypto. I'll try to rename it.